e2e/node/acp-v3.test.ts

```diff
@@ -0,0 +1,271 @@
+/**
+ * Copyright 2022 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import {
+  jest,
+  describe,
+  it,
+  expect,
+  beforeEach,
+  afterEach,
+} from "@jest/globals";
+
+import { Session } from "@inrupt/solid-client-authn-node";
+import { config } from "dotenv-flow";
+import {
+  getSolidDataset,
+  setThing,
+  saveSolidDatasetAt,
+  overwriteFile,
+  getFile,
+  getSourceUrl,
+  deleteFile,
+  createSolidDataset,
+  deleteSolidDataset,
+  UrlString,
+  acp_v3 as acp,
+  FetchError,
+} from "../../src/index";
+import { getTestingEnvironment, TestingEnvironment } from "./util/getTestingEnvironment";
+import { supportsAcp } from "./util/supportsAcp";
+import { getSession } from "./util/getSession";
+
+let env: TestingEnvironment;
+
+beforeAll(() => {
+  config({
+    path: __dirname,
+    // Disable warning messages in CI
+    silent: process.env.CI === "true",
+  });
+  env = getTestingEnvironment();
+});
+
+describe(`Authenticated end-to-end ACP V3 tests against environment ${env.environment}`, () => {
+  if (!supportsAcp(env.pod)) {
+    return;
+  }
+
+  let options: any;
+  let session: Session;
+  let sessionResource: string;
+
+  beforeEach(async () => {
+    session = await getSession(env);
+    sessionResource = `${env.pod}solid-client-tests/node/acp-v3-test-dataset-${session.info.sessionId}`;
```

Comment:
If possible, could the URL constructor be used here instead of plain string templates ?